core/embed/projects/prodtest/cmd/prodtest_tropic.c

```diff
@@ -402,6 +404,119 @@ static void prodtest_tropic_certfido_write(cli_t* cli) {
   cli_ok(cli, "");
 }
 
+static void prodtest_tropic_update_fw(cli_t* cli) {
+#define FW_APP_UPDATE_BANK FW_BANK_FW1
+#define FW_SPECT_UPDATE_BANK FW_BANK_SPECT1
+
+  if (cli_arg_count(cli) > 0) {
+    cli_error_arg_count(cli);
+    return;
+  }
+
+  lt_handle_t* h = tropic_get_handle();
+
+  lt_chip_id_t chip_id = {0};
+  if (lt_get_info_chip_id(h, &chip_id) != LT_OK) {
+    cli_error(cli, CLI_ERROR, "Unable to get CHIP ID");
+    return;
+  }
+
+  cli_trace(cli, "Silicon revision: %c%c%c%c", chip_id.silicon_rev[0],
+            chip_id.silicon_rev[1], chip_id.silicon_rev[2],
+            chip_id.silicon_rev[3]);
+
+#ifdef ABAB
+  if (strncmp((char*)chip_id.silicon_rev, "ABAB", 4) != 0) {
+    cli_error(cli, CLI_ERROR, "Wrong tropic chip silicon revision");
+    return;
+  }
+#else
+  cli_error(cli, CLI_ERROR, "Tropic chip silicon revision not set");
+  return;
+#endif
+
+  lt_ret_t ret;
+
+  // Reused variable
+  uint8_t fw_ver[LT_L2_GET_INFO_RISCV_FW_SIZE] = {0};
+
+  // For firmware update chip must be rebooted into MAINTENANCE mode.
```

Comment:
A suggestion: I think the code would be more readable if you consistently used
```c
if (!is_everythink_ok()) {
  cli_error(cli, CLI_ERROR, "message");
  return;
}
do_something();
```

instead of

```c
if (is_everythink_ok()) {
  do_somethinkg();
} else {
  cli_error(cli, CLI_ERROR, "message");
  return;
}
```

Reply:
refactored in https://github.com/trezor/trezor-firmware/commit/31b417dbe80a3cd08181cb65e902b19b6f466bb3

the style completely makes sense to me